### PR TITLE
Issue #50 snake resolver

### PR DESCRIPTION
**Pull request by benson-basis**
Have a look? It fails my own OSGi test at the moment, I haven't sorted out why.

### CODE REVIEW

File: src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLParser.java [hunk 1]
```diff
@@ -439,53 +443,43 @@ protected JsonToken _decodeScalar(ScalarEvent scalar)
...
+                    return B ? JsonToken.VALUE_TRUE : JsonToken.VALUE_FALSE;
                 }
+            } else if (nodeTag == Tag.NULL) {
+                return JsonToken.VALUE_NULL;
+            } else {
+                // what to do with timestamp and binary and merge etc.
+                return JsonToken.VALUE_STRING;
             }
         } else { // yes, got type tag
+            if (typeTag.startsWith("tag:yaml.org,2002:")) {
```
Comment (cowtowncoder) [inline src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLParser.java]: Is this a well-known id? Could you add a brief comment explaining what/why (or link to spec)?

File: src/test/java/com/fasterxml/jackson/dataformat/yaml/SimpleParseTest.java [hunk 2]
```diff
@@ -325,4 +325,38 @@ public void testObjectWithNumbers() throws Exception
...
+        assertToken(JsonToken.END_ARRAY, jp.nextToken());
+        assertToken(JsonToken.END_OBJECT, jp.nextToken());
+        assertNull(jp.nextToken());
+
+        jp.close();
+    }
+
+    public void testTildeNulls() throws Exception
+    {
+        String YAML = "nulls: [~ ]";
```
Comment (cowtowncoder) [inline src/test/java/com/fasterxml/jackson/dataformat/yaml/SimpleParseTest.java]: Wow. Wasn't aware of ... snake operator? A link/comment would be useful here as well.

**Comment by cowtowncoder** [comment]
I assume OSGi issue can be resolved, as you are more of an export than I am, so will merge.

File: src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLParser.java (hunk 1 above)
Comment (benson-basis) [inline src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLParser.java]: You went ahead and merged. I think this is all in the YAML 1.1 spec; see http://yaml.org/spec/1.1/#id861700, see section 2.4, see 3.1.1. The tags like this are scattered all over the doc, and never in a list.

File: src/test/java/com/fasterxml/jackson/dataformat/yaml/SimpleParseTest.java (hunk 2 above)
Comment (benson-basis) [inline src/test/java/com/fasterxml/jackson/dataformat/yaml/SimpleParseTest.java]: See links above. Do you want another PR with more comments?

Comment (cowtowncoder) [inline src/test/java/com/fasterxml/jackson/dataformat/yaml/SimpleParseTest.java]: If you can add a suggested change here I'll cut'n paste. A link to spec would be all I need if you know it off-hand?

Comment (benson-basis) [inline src/test/java/com/fasterxml/jackson/dataformat/yaml/SimpleParseTest.java]: I put a bunch of spec links into a comment, didn't you see them?

Comment (cowtowncoder) [inline src/test/java/com/fasterxml/jackson/dataformat/yaml/SimpleParseTest.java]: I must have been reading updates in somewhat arbitrary order. I do see ref to http://yaml.org/spec/1.1/#id861700 thanks!
